modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/DataStreamIT.java

```diff
@@ -1791,7 +1791,8 @@ public ClusterState execute(ClusterState currentState) throws Exception {
                         original.getLifecycle(),
                         original.isFailureStore(),
                         original.getFailureIndices(),
-                        null
+                        original.rolloverOnWrite(),
+                        original.getAutoShardingEvent()
```

Comment:
There are a couple of changes like this, where I also copy the auto-sharding event rather than passing null (only in tests). From what I could see, none of these tests actually cared about the auto-sharding event, so we should just copy the original value. This will also simplify the use of the `Builder` class that's being worked on in another PR.